src/app/lib/prisma.ts

```diff
@@ -1,35 +1,62 @@
 import { PrismaClient } from "@prisma/client";
+import { time_this } from "@/app/lib/logging";
 
 declare global {
   var prisma: PrismaClient;
 }
 
 let prisma: PrismaClient;
 
+// Logging middleware
+const makePrismaClient = () => {
+  const execRaw = async (query: (args: any) => Promise<any>, args:any, operation: string) => {
+    return await time_this(
+      async () => await query(args),
+      {
+        operation,
+        args,
+      }
+    );
+  };
+  return new PrismaClient().$extends({
+    query: {
+      $allModels: {
+        async $allOperations({ operation, model, args, query }) {
+          return await time_this(
+            async () => await query(args),
+            {
+              model,
+              operation,
+              args,
+            }
+          )
+        },
+      },
+      async $queryRaw({ args, query, operation }) {
+        return await execRaw(query, args, operation);
+      },
+      async $executeRaw({ args, query, operation }) {
+        return await execRaw(query, args, operation);
+      },
+      async $queryRawUnsafe({ args, query, operation }) {
+        return await execRaw(query, args, operation);
+      },
+      async $executeRawUnsafe({ args, query, operation }) {
+        return await execRaw(query, args, operation);
+      },
+    },
+  });
+};
+
 if (process.env.NODE_ENV === "production") {
-  prisma = new PrismaClient();
+  prisma = makePrismaClient() as PrismaClient;
```

Comment:
Does this also run in prod?

Reply:
I think that line would. I didn't trace out what was going on with the conditionals on the client instantiation, so I kept the environment checks the same.